src/hera/workflows/task.py

```diff
@@ -55,6 +104,9 @@ class Task(
     dependencies: Optional[List[str]] = None
     depends: Optional[str] = None
 
+    _default_next_operator: ClassVar[Operator] = Operator.and_
+    _default_next_on: ClassVar[OnType] = None
```

Comment:
We store the result of `_normalize_on` here, which is not the fully generic `OnType`. Storing an unknown iterable in this field would potentially be a bug, since iterables can be altered by iterating them, so we should specify the exact type to avoid a risk of issues in future.
```suggestion
    _default_next_on: ClassVar[Optional[List[TaskResult]]] = None
```